openhcl/virt_mshv_vtl/src/processor/snp/mod.rs

```diff
@@ -1836,10 +1849,11 @@ impl AccessVpState for UhVpStateAccess<'_, '_, SnpBacked> {
             vp::MpState::Running => (false, false),
             vp::MpState::WaitForSipi => (false, true),
             vp::MpState::Halted => (true, false),
-            vp::MpState::Idle => (false, false), // TODO SNP: idle support
+            vp::MpState::Idle => (false, false),
         };
         let lapic = &mut self.vp.backing.lapics[self.vtl];
         lapic.halted = halted;
+        lapic.idle = mp_state == vp::MpState::Idle;
```

Comment:
Maybe move this into the match above? Would make things clearer IMO

Reply:
Sure. I also wonder whether something like the following would be clearer:
```rust
    fn set_activity(&mut self, value: &vp::Activity) -> Result<(), Self::Error> {
        let &vp::Activity {
            mp_state,
            nmi_pending,
            nmi_masked: _,           // TODO SNP
            interrupt_shadow: _,     // TODO SNP
            pending_event: _,        // TODO SNP
            pending_interruption: _, // TODO SNP
        } = value;
        let lapic = &mut self.vp.backing.lapics[self.vtl];
        lapic.halted = mp_state == vp::MpState::Halted;
        lapic.idle = mp_state == vp::MpState::Idle;
        lapic.startup_suspend = mp_state == vp::MpState::WaitForSipi;
        lapic.nmi_pending = nmi_pending;
        Ok(())
    }
    ```
Any opinion?

Reply:
I think that'd be fine too, so long as they're unified